kratos/solving_strategies/builder_and_solvers/explicit_builder.h

```diff
@@ -0,0 +1,819 @@
+     * @brief Function to perform the build of the RHS.
+     * The vector could be sized as the total number of dofs or as the number of unrestrained ones
+     * @param rModelPart The model part to compute
+     */
+    virtual void BuildRHSNoDirichlet(ModelPart& rModelPart)
+    {
+        KRATOS_TRY
+
+        // Initialize the reaction (residual)
+        InitializeDofSetReactions();
+
+        // Gets the array of elements, conditions and constraints from the modeler
+        const auto &r_elements_array = rModelPart.Elements();
+        const auto &r_conditions_array = rModelPart.Conditions();
+        const int n_elems = static_cast<int>(r_elements_array.size());
+        const int n_conds = static_cast<int>(r_conditions_array.size());
+
+        const auto& r_process_info = rModelPart.GetProcessInfo();
+
+#pragma omp parallel firstprivate(n_elems, n_conds)
+        {
+#pragma omp for schedule(guided, 512) nowait
+            // Assemble all elements
+            for (int i_elem = 0; i_elem < n_elems; ++i_elem) {
+                auto it_elem = r_elements_array.begin() + i_elem;
+                // Detect if the element is active or not. If the user did not make any choice the element is active by default
+                // TODO: We will require to update this as soon as we remove the mIsDefined from the Flags
+                bool element_is_active = true;
+                if (it_elem->IsDefined(ACTIVE)) {
+                    element_is_active = it_elem->Is(ACTIVE);
+                }
+
+                if (element_is_active) {
+                    // Calculate elemental explicit residual contribution
+                    // The explicit builder and solver assumes that the residual contribution is assembled in the REACTION variables
+                    it_elem->AddExplicitContribution(r_process_info);
+                }
+            }
+
+            // Assemble all conditions
+#pragma omp for schedule(guided, 512)
+            for (int i_cond = 0; i_cond < n_conds; ++i_cond) {
+                auto it_cond = r_conditions_array.begin() + i_cond;
+                // Detect if the condition is active or not. If the user did not make any choice the condition is active by default
+                // TODO: We will require to update this as soon as we remove the mIsDefined from the Flags
+                bool condition_is_active = true;
+                if (it_cond->IsDefined(ACTIVE)) {
+                    condition_is_active = it_cond->Is(ACTIVE);
+                }
+
+                if (condition_is_active) {
+                    // Calculate condition explicit residual contribution
+                    // The explicit builder and solver assumes that the residual contribution is assembled in the REACTION variables
+                    it_cond->AddExplicitContribution(r_process_info);
+                }
+            }
+        }
+
+        KRATOS_CATCH("")
+    }
+
+    // TODO: Define this once the MPC work in the explicit framework
```

Comment:
I added something to add MPC contribution in explicit, please check this:

https://github.com/KratosMultiphysics/Kratos/blob/de0dca1aaf55019f040133b59a44f05d74f1e309/applications/StructuralMechanicsApplication/custom_strategies/custom_strategies/mechanical_explicit_strategy.hpp#L481

Reply:
Should be as easy as adding this?

Reply:
As easy as copy paste the piece of code I attached you

Reply:
Cool. I'll add it. Thanks @loumalouomega !

Reply:
Done. I added to the strategy too. Can you please check and approve if it's correct?